### DISCUSSION TITLE
Can I check the modification of the display in ProjectUpdatePostEvent?

### DISCUSSION THREAD

**Discussion by seleondevel** [kimai/kimai#4852] [Plugins]
![스크린샷 2024-05-13 162948](https://github.com/kimai/kimai/assets/165639563/412dc655-2d4f-4aa5-8e2a-f1f321884fd1)

From editing the displayed project
If you turn off the display and save it, you want to implement a function to send a notification email that the project is over.
Can I check the modification of the display in ProjectUpdatePostEvent?

**Comment by kevinpapst** [comment]
No you can't. You need to create a Doctrine subscriber like this:
https://github.com/kimai/kimai/blob/main/src/Doctrine/TimesheetSubscriber.php#L61

**Reply by seleondevel** [reply]
To send a notification that the project is over, which event should I implement as a Doctorine subscriber?

**Reply by seleondevel** [reply]
public function updateProject(Project $project): Project
    {
        $this->validateProject($project);

        $this->dispatcher->dispatch(new ProjectUpdatePreEvent($project));
        $this->repository->saveProject($project);
        $this->dispatcher->dispatch(new ProjectUpdatePostEvent($project));

        return $project;
    }

 public function onProjectUpdatePostEvent(ProjectUpdatePostEvent $event){
        $this->logger->info(sprintf('ProjectUpdatePostEvent!!'));
        dd($event);
    }
    
public static function getSubscribedEvents()
    {
        return [        
            ProjectUpdatePostEvent::class => 'onProjectUpdatePostEvent',            
];
    }
}

    
Why doesn't anything run on ProjectUpdatePostEvent when you turn off visibility in project editing?

**Reply by seleondevel** [reply]
When can I get the ProjectUpdatePostEvent notification?

**Reply by kevinpapst** [reply]
This is open source software, not a paid Q&A forum. 

If something doesn't work as expected, you are welcome to debug it and post a solution or a bug fix to change the behavior.
